hbase-client/src/test/java/org/apache/hadoop/hbase/security/TestSaslUtil.java

```diff
@@ -61,4 +62,118 @@ public void testInitSaslProperties() {
     props = SaslUtil.initSaslProperties("");
     assertEquals("auth", props.get(Sasl.QOP));
   }
+
+  @Test
+  public void testVerifyQop() {
+    String nullQop = null;
+    String authentication = "auth";
+    String integrity = "auth-int";
+    String confidentality = "auth-conf";
+    String anyQop = "auth-conf,auth-int,auth";
+
+    // Empty requested, got empty
+    try {
+      SaslUtil.verifyNegotiatedQop(nullQop, nullQop);
+    } catch (Exception e) {
+      fail("Should not have thrown exception");
```

Comment:
Just let the exception be thrown? In this way we could see more detailed failure message...

Reply:
Done.